alluxiofs/core.py

```diff
@@ -224,123 +245,110 @@ def cat_file(self, path, start=None, end=None, **kwargs):
         path = self.unstrip_protocol(path)
         return self.alluxio.read_range(path, start, length)
 
-    def ukey(self, *args, **kwargs):
-        if self.fs:
-            self.fs.ukey(*args, **kwargs)
-        else:
-            raise NotImplementedError
-
-    def mkdir(self, *args, **kwargs):
-        if self.fs:
-            self.fs.mkdir(*args, **kwargs)
-        else:
-            raise NotImplementedError
-
-    def makedirs(self, *args, **kwargs):
-        if self.fs:
-            self.fs.makedirs(*args, **kwargs)
-        else:
-            raise NotImplementedError
-
-    def rm(self, *args, **kwargs):
-        if self.fs:
-            self.fs.rm(*args, **kwargs)
-        else:
-            raise NotImplementedError
+    def fallback_handler(func):
+        @wraps(func)
+        def fallback_wrapper(self, path, *args, **kwargs):
+            if self.fs:
+                fs_method = getattr(self.fs, func.__name__, None)
+                if fs_method:
+                    return fs_method(
+                        self._strip_alluxio_protocol(path), *args, **kwargs
+                    )
+                raise NotImplementedError(
+                    f"The method {func.__name__} is not implemented in the underlying filesystem."
+                )
+            else:
+                raise NotImplementedError(
+                    "Underlying filesystem is not available."
+                )
```

Comment:
why dont we merge this with alluxio_with_fallback_handler?
coz those not implemented in fsspec is not implemented in alluxio is it? if alluxio throws unimplemented error then we check self.fs (fsspec ufs) and then if it still isnt present it throws notimplmented right?